build/common/installer/scripts/tomlparser.rb

```diff
@@ -30,13 +30,143 @@
+  "mem_buf_limit" => "10m",
+  "buffer_chunk_size" => "1m",
+  "buffer_max_size" => "5m",
+  "throttle_rate" => "1000",
+  "throttle_window" => "300",
+  "throttle_interval" => "1s",
+  "out_forward_worker_count" => "10",
+  "out_forward_retry_limit" => "10",
+  "out_forward_storage_total_limit_size" => "2G",
+  "out_forward_require_ack_response" => "false"
+}
+
+def is_windows?()
+  return !@os_type.nil? && !@os_type.empty? && @os_type.strip.casecmp("windows") == 0
+end
 
-
-if !@os_type.nil? && !@os_type.empty? && @os_type.strip.casecmp("windows") == 0
+if is_windows?()
   @containerLogsRoute = "v1" # default is v1 for windows until windows agent integrates windows ama
   # This path format is necessary for fluent-bit in windows
   @logTailPath = "C:\\var\\log\\containers\\*.log"
 end
+
+def is_number?(value)
+  true if Integer(value) rescue false
+end
+
+def is_valid_number?(value)
+  return !value.nil? && is_number?(value) && value.to_i > 0
+end
+
+def updateDefaultConfigSetting(configEntryName, configValue)
+  begin
+    if !configValue.nil? && !configValue.empty?
+      if is_number?(configValue) && configValue.to_i > 0
+        @default_settings[configEntryName] = configValue
+      else
+        @default_settings[configEntryName] = configValue.to_s
+      end
+    end
+  rescue => errorStr
+    ConfigParseErrorLogger.logError("Exception while updateDefaultConfigSetting- #{errorStr}")
+  end
+end
+
+def getNamespaceSettingsConfigEntryValue(configSettings, configEntryName)
+  value = @default_settings[configEntryName]
+  begin
+    if !configSettings.nil? && !configSettings[configEntryName].nil? && !configSettings[configEntryName].empty?
+      value = configSettings[configEntryName]
+    end
+  rescue => errorStr
+    ConfigParseErrorLogger.logError("Exception while getting getConfigEntryValue - #{errorStr}")
+  end
+  return value
+end
+
+def clearTemplateFile(templatefilePath)
+  begin
+    if File.file?(templatefilePath)
+      File.open(templatefilePath, 'r+') do |file|
+        file.truncate(0) # clear the file
+      end
+    end
+  rescue => exception
+    ConfigParseErrorLogger.logError("clearTemplateFile: Exception while clearing template file: #{exception}")
+  end
+end
+
+def getTenantTemplateFilePath()
+  templatefilePath = "/etc/opt/microsoft/docker-cimprov/fluent-bit-azmon-logs_tenant.conf"
+   if is_windows?()
+    templatefilePath =  "C:\\etc\\fluent-bit\\fluent-bit-azmon-logs_tenant.conf"
+   end
+  return templatefilePath
+end
+
+def getTenantFilePath(tenant_namespace)
+  tenant_file_path = "/etc/opt/microsoft/docker-cimprov/fluent-bit-azmon-logs_tenant_#{tenant_namespace}.conf"
+  if is_windows?()
+    tenant_file_path =  "C:\\etc\\fluent-bit\\fluent-bit-azmon-logs_tenant_#{tenant_namespace}.conf"
+  end
+  return tenant_file_path
+end
+
+def generateAzMonMultiTenantNamespaceConfig
+   templatefilePath = getTenantTemplateFilePath()
+   begin
+    @azMonMultiTenantNamespaces.each do |namespace|
```

Comment:
if namespaces is large, will this introduce heave cpu or memory usage? Do we need to consider concurrent processing files?